src/HotChocolate/AspNetCore/src/AspNetCore/Types/FileType.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using HotChocolate.Language;
+using Microsoft.AspNetCore.Http;
+
+namespace HotChocolate.Types
+{
+    // todo: This is not fully implemented
+    public class FileValueNode : IValueNode<IFormFile>
+    {
+        private IFormFile _value;
+
+        public FileValueNode(object value)
+        {
+            if (value is null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            if (value is IFormFile formFileValue)
+            {
+                _value = formFileValue;
+            }
+            else
+            {
+                throw new Exception("Value is not IFormFile");
+            }
+        }
+
+        public IFormFile Value => _value;
+
+        public SyntaxKind Kind => SyntaxKind.Argument;
+
+        public Language.Location? Location => throw new NotImplementedException();
+
+        object? IValueNode.Value => _value;
+
+        public bool Equals(IValueNode? other)
+        {
+            if (other == null)
+                return false;
+
+            return other.Value != Value;
+        }
+
+        public IEnumerable<ISyntaxNode> GetNodes()
+        {
+            throw new NotImplementedException();
+        }
+
+        public string ToString(bool indented)
+        {
+            return _value?.Name ?? "";
+        }
+    }
+
+    // todo: This is not fully implemented
+    public class FileType : ScalarType<IFormFile>
```

Comment:
This one will be quite interesting ... have to think about how we can tackle this :)